### ISSUE TITLE
master build tests broken?

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#687]
View the changeset: https://github.com/coredns/coredns/compare/e261ac1a6e64...30ecb83dce02

View the full build log and details: https://travis-ci.org/coredns/coredns/builds/238306443?utm_source=email&utm_medium=notification

~~~
=== RUN   TestKubernetesIntegration
--- FAIL: TestKubernetesIntegration (1.04s)
	kubernetes_test.go:460: Expected 7 answers but got 6 for query *.test-1.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 7 answers but got 6 for query any.test-1.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 7 answers but got 6 for query *.*.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 2 answers but got 1 for query ext-svc.test-1.svc.cluster.local., 1
=== RUN   TestKubernetesIntegrationPodsInsecure
--- PASS: TestKubernetesIntegrationPodsInsecure (1.00s)
=== RUN   TestKubernetesIntegrationPodsVerified
--- PASS: TestKubernetesIntegrationPodsVerified (1.02s)
=== RUN   TestKubernetesIntegrationCidrReverseZone
--- PASS: TestKubernetesIntegrationCidrReverseZone (1.03s)
=== RUN   TestKubernetesIntegrationPartialCidrReverseZone
--- PASS: TestKubernetesIntegrationPartialCidrReverseZone (1.01s)
=== RUN   TestKubernetesIntegrationAllNSExposed
--- PASS: TestKubernetesIntegrationAllNSExposed (1.01s)
=== RUN   TestKubernetesIntegrationFallthrough
--- FAIL: TestKubernetesIntegrationFallthrough (1.04s)
	kubernetes_test.go:460: Expected 7 answers but got 6 for query *.test-1.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 7 answers but got 6 for query any.test-1.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 7 answers but got 6 for query *.*.svc.cluster.local., 1
	kubernetes_test.go:460: Expected 2 answers but got 1 for query ext-svc.test-1.svc.cluster.local., 1
=== RUN   TestMetricsServer
--- PASS: TestMetricsServer (0.01s)
~~~

**Comment by johnbelamaric** [comment]
@chrisohaver you saw something like this while doing the external CNAME stuff?

**Comment by chrisohaver** [comment]
I think this happens when services aren't ready when the test starts.
I can add a smarter wait, which waits for all services to be ready.

**Comment by chrisohaver** [comment]
fixed with PR #689
